contracts/contracts/core/vault/src/main.sw

```diff
@@ -1187,355 +745,151 @@ fn _get_after_fee_amount(
-        fee_basis_points_0
-    } else {
-        fee_basis_points_1
-    }
-}
-
-#[storage(read)]
-fn _get_redemption_amount(asset: AssetId, rusd_amount: u256) -> u256 {
-    let price = _get_max_price(asset);
-    let redemption_amount = rusd_amount * PRICE_PRECISION / price;
-
-    let rusd = storage.rusd.read();
-    _adjust_for_decimals(redemption_amount, rusd, asset)
-}
-
-#[storage(read)]
-fn _get_target_rusd_amount(asset: AssetId) -> u256 {
-    let supply = abi(RUSD, storage.rusd_contr.read().into()).total_rusd_supply();
-     let weight = storage.asset_weights.get(asset).try_read().unwrap_or(0);
-
-    (weight * supply / storage.total_asset_weights.read()).as_u256()
-}
-
-#[storage(read)]
-fn _adjust_for_decimals(
-    amount: u256, 
-    asset_div: AssetId, 
-    asset_mul: AssetId
-) -> u256 {
-    let rusd = storage.rusd.read();
-    let decimals_div = if asset_div == rusd {
-        RUSD_DECIMALS
-    } else {
-        storage.asset_decimals.get(asset_div).try_read().unwrap_or(0)
-    };
-
-    let decimals_mul = if asset_mul == rusd {
-        RUSD_DECIMALS
-    } else {
-        storage.asset_decimals.get(asset_mul).try_read().unwrap_or(0)
-    };
-
-    // this should fail if there's some weird stack overflow error
-    require(
-        decimals_div != 0 || decimals_mul != 0,
-        Error::VaultDecimalsAreZero
-    );
-
-    amount * 10.pow(decimals_mul).as_u256() / 10.pow(decimals_div).as_u256()
-}
-
-#[storage(read)]
-fn _asset_to_usd_min(asset: AssetId, asset_amount: u256) -> u256 {
-    if asset_amount == 0 {
-        return 0;
-    }
-
-    let price = _get_min_price(asset);
-    let decimals = storage.asset_decimals.get(asset).try_read().unwrap_or(0);
-
-    (asset_amount * price) / 10.pow(decimals).as_u256()
-}
-
-#[storage(read)]
-fn _usd_to_asset_max(asset: AssetId, usd_amount: u256) -> u256 {
-    if usd_amount == 0 {
-        return 0;
-    }
-
-    // @notice this is CORRECT (asset_max -> get_min_price)
-    let price = _get_min_price(asset);
-
-    _usd_to_asset(asset, usd_amount, price)
-}
-
-#[storage(read)]
-fn _usd_to_asset_min(asset: AssetId, usd_amount: u256) -> u256 {
-    if usd_amount == 0 {
-        return 0;
-    }
-
-    // @notice this is CORRECT (asset_min -> get_max_price)
-    let price = _get_max_price(asset);
-
-    _usd_to_asset(asset, usd_amount, price)
-}
-
-#[storage(read)]
-fn _usd_to_asset(asset: AssetId, usd_amount: u256, price: u256) -> u256 {
-    require(price != 0, Error::VaultPriceQueriedIsZero);
-
-    if usd_amount == 0 {
-        return 0;
-    }
-
-    let decimals = storage.asset_decimals.get(asset).try_read().unwrap_or(0);
-
-    (usd_amount * 10.pow(decimals).as_u256()) / price
-}
-
-#[storage(read)]
-fn _get_max_price(asset: AssetId) -> u256 {
-    let vault_pricefeed = abi(VaultPricefeed, storage.pricefeed_provider.read().into());
-    vault_pricefeed.get_price(
-        asset, 
-        true,
-    )
+/// price in the collateral
+fn _get_max_price(asset: b256) -> u256 {
```

Comment:
`_get_max_price` and `_get_min_price` are exactly the same.
I guess they will have something to do with spread in the future. IF that's the case, then leave a comment why they are the same now.